config/routes.rb

```diff
@@ -15,5 +15,10 @@
         get '/' => 'node#show'
       end
     end
+    scope '/plan' do
+      get '/' => 'plan#index'
+      post '/' => 'plan#create'
+      get '/:id' => 'plan#show'
+    end
```

Comment:
The way you have setup the _routes.rb_ file doesn't adhere to the Rails philosophy. What you are doing works but it is effectively what Rails will do for you should you configure it the way they define in the docs. This is a nice implementation of an Express-like approach in Rails though! 

- [CRUD Verbs and Actions](http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions)
- [Controller Namespaces and Routing](http://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing)
- [Restricting the Routes Created](http://guides.rubyonrails.org/routing.html#restricting-the-routes-created)

Reply:
This black boxing is very hard to get used to. I'll fix that in the next version of this PR.